com.unity.shadergraph/Editor/ShaderFoundry/Provider/LegacyTemplateProvider.cs

```diff
@@ -86,17 +87,49 @@ void AddTemplateTag(string tagName, string tagValue)
             foreach (var pass in subShaderDescriptor.passes)
             {
                 var legacyPassDescriptor = pass.descriptor;
-                var passBuilder = new TemplatePass.Builder(Container);
-                passBuilder.ReferenceName = legacyPassDescriptor.referenceName;
-                passBuilder.DisplayName = legacyPassDescriptor.displayName;
-                passBuilder.SetPassIdentifier((uint)subShaderIndex, (uint)subPassIndex);
-                ++subPassIndex;
+                var templatePass = BuildTemplatePass(legacyPassDescriptor, subShaderIndex, subPassIndex, vertexCustomizationPoint, surfaceCustomizationPoint);
+                subPassIndex++;
+                builder.AddPass(templatePass);
+            }
+
+            builder.AdditionalShaderID = subShaderDescriptor.additionalShaderID;
+            builder.ShaderFallback = subShaderDescriptor.shaderFallback ?? "Hidden/Shader Graph/FallbackError";
+
+            if (subShaderDescriptor.usePassList != null)
+            {
+                foreach (var usePass in subShaderDescriptor.usePassList)
+                    builder.AddUsePass(usePass);
+            }
 
-                BuildLegacyTemplateEntryPoints(legacyPassDescriptor, passBuilder, vertexCustomizationPoint, surfaceCustomizationPoint);
+            if (subShaderDescriptor.shaderDependencies != null)
+            {
+                foreach (var dependency in subShaderDescriptor.shaderDependencies)
+                    builder.AddShaderDependency(dependency.dependencyName, dependency.shaderName);
+            }
+
+            string shaderGUI = null;
+            string rpAssetType = null;
+
+            // TODO: we might just be able to get rid of subShaderDescriptor.shaderCustomEditor,
+            // I don't believe anyone is still using it...
+            if (!string.IsNullOrWhiteSpace(subShaderDescriptor.shaderCustomEditor))
+            {
+                shaderGUI = subShaderDescriptor.shaderCustomEditor;
+                rpAssetType = null;
+            }
 
-                builder.AddPass(passBuilder.Build());
+            if (subShaderDescriptor.shaderCustomEditors != null)
+            {
+                foreach (var customEditor in subShaderDescriptor.shaderCustomEditors)
```

Comment:
Last one in the list wins?  Is that the behavior we want?  (not saying it's wrong, just asking)

Reply:
Yeah.  None of the existing legacy targets are providing more than one anyways...   :)